sigoptlite/sigoptlitetest/endpoints/test_best_assignments.py

```diff
@@ -128,107 +160,155 @@ def test_multimetric_thresholds(self, threshold_y1, threshold_y2):
+        values=[dict(name="y1", value=numpy.random.rand()), dict(name="y2", value=numpy.random.rand())],
+      )
+
+    best_assignments_list = list(self.conn.experiments(e.id).best_assignments().fetch().iterate_pages())
+    self.assert_best_assignments_are_sorted(e, best_assignments_list)
+    assert len(best_assignments_list) > 1
+
+    for best_assignment in best_assignments_list:
+      for i, metric in enumerate(experiment_meta["metrics"]):
+        assert metric["name"] == best_assignment.values[i].name
+        if metric["objective"] == "maximize":
+          assert best_assignment.values[i].value >= metric["threshold"]
+        if metric["objective"] == "minimize":
+          assert best_assignment.values[i].value <= metric["threshold"]
 
   def test_multitask(self):
     experiment_meta = self.get_experiment_feature("multitask")
-    experiment = LocalExperimentBuilder(experiment_meta)
-    observations = []
-    for _ in range(20):
-      observation_with_task = self.make_observation(
-        experiment=experiment,
-        assignments=self.make_random_suggestions(experiment)[0].assignments,
-        values=[dict(name="y1", value=numpy.random.rand())],
-        task=numpy.random.choice(experiment_meta["tasks"]),
+    experiment_meta["type"] = "random"
+    e = self.conn.experiments().create(**experiment_meta)
+
+    wrong_optimal_value = 200
+    num_observations = 25
+    for _ in range(num_observations):
+      suggestion = self.conn.experiments(e.id).suggestions().create()
+      if suggestion.task.cost < 1:
+        values = [dict(name="y1", value=wrong_optimal_value)]
+      else:
+        values = [dict(name="y1", value=numpy.random.rand())]
+      self.conn.experiments(e.id).observations().create(
+        suggestion=suggestion.id,
+        values=values,
+        task=suggestion.task,
       )
-      observations.append(observation_with_task)
 
-    best_assignments_logger = BestAssignmentsLogger(experiment)
-    best_observations_from_logger = best_assignments_logger.fetch(observations)
-    assert len(best_observations_from_logger) == 1
-    assert best_observations_from_logger[0]["task"]["cost"] == 1
-    assert best_observations_from_logger[0]["task"]["name"] == "expensive"
+    best_assignments_list = list(self.conn.experiments(e.id).best_assignments().fetch().iterate_pages())
+    assert len(best_assignments_list) == 1
+    assert best_assignments_list[0].values[0].value <= 1
+
+  @pytest.mark.parametrize(
+    "metrics",
+    [
+      [dict(name="y1", objective="maximize", threshold=1), dict(name="y2", objective="maximize", threshold=1)],
+      [
+        dict(name="y1", objective="maximize", strategy="constraint", threshold=1),
+        dict(name="y2", objective="minimize", strategy="optimize"),
+      ],
+      [
+        dict(name="y1", objective="maximize", strategy="constraint", threshold=1),
+        dict(name="y2", objective="minimize", strategy="constraint", threshold=1),
+      ],
+    ],
```

Comment:
These test check are a bit cheeky; they depend on that you are generating y1, y2 in [0, 1]. 

I think a better way of setting up this test is to create the experiments as defined in the feature constants and then manually create a few observations with values that violates 1 or all of the thresholds (which you can access via either the meta or the experiment object).